Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/DecoratorBased/BackpackEventBusDecorator.cs

```diff
@@ -0,0 +1,122 @@
+﻿using DCL.AvatarRendering.Emotes;
+using DCL.AvatarRendering.Wearables.Components;
+using DCL.Backpack.BackpackBus;
+using DCL.CharacterPreview;
+using DCL.UI;
+using Segment.Serialization;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.PerformanceAndDiagnostics.Analytics
+{
+    public class BackpackEventBusAnalyticsDecorator : IBackpackEventBus
+    {
+        private readonly IBackpackEventBus core;
+        private readonly IAnalyticsController analytics;
+
+        public event Action<IWearable> SelectWearableEvent;
+        public event Action<IWearable> EquipWearableEvent;
+        public event Action<IWearable> UnEquipWearableEvent;
+        public event Action<int, IEmote, bool> EquipEmoteEvent;
+        public event Action<int, IEmote> UnEquipEmoteEvent;
+        public event Action<int> EmoteSlotSelectEvent;
+        public event Action<IEmote> SelectEmoteEvent;
+        public event Action<IReadOnlyCollection<string>> ForceRenderEvent;
+        public event Action<string> FilterCategoryEvent;
+        public event Action<string> SearchEvent;
+        public event Action<BackpackSections> ChangedBackpackSectionEvent;
+        public event Action<Color, string> ChangeColorEvent;
+        public event Action UnEquipAllEvent;
+        public event Action PublishProfileEvent;
+        public event Action<AvatarWearableCategoryEnum> FilterCategoryByEnumEvent;
+        public event Action DeactivateEvent;
+
+        public BackpackEventBusAnalyticsDecorator(IBackpackEventBus core, IAnalyticsController analytics)
+        {
+            this.core = core;
+            this.analytics = analytics;
+
+            core.EquipEmoteEvent += (slot, emote, manuallyEquipped) =>
```

Comment:
I think its better to subscribe with a private function, referring to the `analytics` as a field. By making an anonymous function it will have to keep this context in memory, so it is reconstructed when the event is triggered